--- a/usr/sbin/start_pms	2020-01-23 16:12:19.000000000 -0800
+++ b/usr/sbin/start_pms	2020-02-04 10:01:47.773486407 -0800
@@ -29,6 +29,9 @@
   export PLEX_MEDIA_SERVER_USER="$PLEX_USER"
 fi
 
+# Activate python virtualenv
+. "${PLEX_MEDIA_SERVER_HOME}"/Resources/Python/bin/activate
+
 # Create AppSuppDir if not present
 if [ ! -d "$PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR" ]; then
   mkdir -p "$PLEX_MEDIA_SERVER_APPLICATION_SUPPORT_DIR"
